src/stream/src/common/table/state_table.rs

```diff
@@ -1107,7 +1107,7 @@ impl<S: StateStore, W: WatermarkBufferStrategy> StateTable<S, W> {
         }
         let prefix_hint = {
             if self.prefix_hint_len == 0 || self.prefix_hint_len > pk_prefix.len() {
-                panic!();
+                None
```

Comment:
`may_exist` should never be called when prifix hint does not exist. The behavior of doing this can be undefined. Panic or report an error here should be correct. PLAT @hzxa21

Reply:
fixed the unit-test by adding `prefix_hint_len` when creating state_table.